torch/_dynamo/utils.py

```diff
@@ -3106,3 +3107,43 @@ def get_user_object_from_id(obj_id):
 def store_user_object_weakref(obj):
     obj_id = id(obj)
     user_obj_id_to_weakref[obj_id] = weakref.ref(obj)
+
+
+class CompileTimeInstructionCounter:
+    _counter: int = 0
+    _id: int = -1
+
+    @classmethod
+    def start(cls) -> None:
+        if cls._id != -1:
+            raise RuntimeError("CompileTimeInstructionCounter is already started")
+        cls._id = i_counter.start()
+
+    @classmethod
+    def end(cls) -> None:
+        cls._counter += i_counter.end(cls._id)
+        cls._id = -1
+
+    @classmethod
+    def clear(cls) -> None:
+        cls._counter = 0
+
+    @classmethod
+    def value(cls) -> int:
+        return cls._counter
+
+
+class CompileTimeInstructionCollector:
```

Comment:
This would probably be easier to read as a contextmanager...

Reply:
you mean something like
```
from contextlib import contextmanager

@contextmanager
def closing(thing):
    try:
        yield thing
    finally:
        thing.close()
 ```

Reply:
yes